include/jemalloc/internal/malloc_io.h

```diff
@@ -90,7 +90,22 @@ malloc_write_fd(int fd, const void *buf, size_t count) {
 }
 
 static inline ssize_t
-malloc_read_fd(int fd, void *buf, size_t count) {
+malloc_write_fd(int fd, const void *buf, size_t count) {
+	size_t bytes_written = 0;
+	while (bytes_written < count) {
+		ssize_t result = malloc_write_fd_syscall(fd,
+		    &((const byte_t *)buf)[bytes_written],
+		    count - bytes_written);
+		if (result == -1) {
+			return -1;
+		}
+		bytes_written += result;
+	}
```

Comment:
`bytes_written` should not be larger than `count`? Can we assert it after the loop ends?

Reply:
`assert`ing anything here would be problematic – `assert` itself calls `malloc_printf` which eventually resolves to a call to this function. If we `assert` because we found an issue, we're likely to start recursing infinitely.